arm-apimanagement/2016-10-10/swagger/apimanagement.json

```diff
@@ -0,0 +1,7693 @@
+            "schema": {
+              "$ref": "#/definitions/SubscriptionContract"
+            },
+            "headers": {              
+              "ETag": {
+                "description": "Current entity state version. Should be treated as opaque and used to make conditional HTTP requests.",
+                "type": "string"
+              }              
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorBodyContract"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "Subscriptions"
+        ],
+        "operationId": "Subscriptions_CreateOrUpdate",
+        "description": "Creates or updates the subscription of specified user to the specified product.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServiceNameParameter"
+          },
+          {
+            "name": "sid",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Identifier of the subscription.",
+            "minLength": 1,
+            "maxLength": 256,
+            "pattern": "^[^*#&+:<>?]+$"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SubscriptionCreateParameters"
+            },
+            "description": "Create parameters."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "The user was successfully subscribed to the product."
+          },
+          "204": {
+            "description": "The user already subscribed to the product."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorBodyContract"
+            }
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "Subscriptions"
+        ],
+        "operationId": "Subscriptions_Update",
+        "description": "Updates the details of a subscription specificied by its identifier.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServiceNameParameter"
+          },
+          {
+            "name": "sid",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Identifier of the subscription."
```

Comment:
Given that sid is being used multiple times you could move it's definition to global "parameters" section like
```json
    "sidParameter": {
      "description": "Identifier of the subscription",
      "name": "sid",
      "in": "path",
      "required": true,
      "type": "string",
      "minLength": 1,
      "maxLength": 256,
      "pattern": "^[^*#&+:<>?]+$",
      "x-ms-parameter-location": "method"
    }
```

and do $ref here.
```json
  {
      "$ref": "#/parameters/sidParameter"
  }
```